packages/apps/sale/src/screens/cart/CatalogVariantScreen.tsx

```diff
@@ -0,0 +1,87 @@
+/*
+ * Axelor Business Solutions
+ *
+ * Copyright (C) 2024 Axelor (<http://axelor.com>).
+ *
+ * This program is free software: you can redistribute it and/or  modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React, {useEffect, useMemo} from 'react';
+import {Screen} from '@axelor/aos-mobile-ui';
+import {
+  SearchListView,
+  displayItemName,
+  useDispatch,
+  useSelector,
+  useTranslator,
+} from '@axelor/aos-mobile-core';
+import {fetchVariantProduct} from '../../features/productSlice';
+import {CartLineActionCard} from '../../components';
+import {fetchActiveCart} from '../../features/cartSlice';
+
+const catalogScanKey = 'sale_catalog_scanKey';
```

Comment:
Just a note for later as this screen will disappear, two screens should not have the same scan key